### PR TITLE
feat: add NotFound(404) and Timeout(408) to jackrabbit

### PR DESCRIPTION
**Pull request by kuryaki**

### CODE REVIEW

File: lib/exchange.js [hunk 1]
```diff
@@ -53,11 +53,45 @@ function exchange(name, type, options) {
...
   return emitter;
 
+  function rpcClient (key, msg, replyTo) {
+    if(!key) {
+      throw new Error('missing rpc method');
+    }
+
+    var timeout = setTimeout(() => {
+      replyTo(new Error('Timeout'))
+    }, 3000);
```
Comment (cilindrox) [inline lib/exchange.js]: What's the deal with the arbitrary `3000`? Shouldn't this be configurable?

File: lib/exchange.js [hunk 2]
```diff
@@ -53,11 +57,53 @@ function exchange(name, type, options) {
...
   });
 
   return emitter;
 
+  function rpcClient (key, msg, options, replyTo) {
+    if(!key) {
+      throw new Error('missing rpc method');
+    }
+
+    if(!replyTo && typeof options === 'function') {
```
Comment (oxymor0n) [inline lib/exchange.js]: now I understand why you wanted kwargs in JS 😛

Comment (oxymor0n) [inline lib/exchange.js]: now I understand why you miss Python's kwargs 😛

Comment (kuryaki) [inline lib/exchange.js]: indeed :D

**Pull request merged**